src/online/unigrid.jl

```diff
@@ -43,3 +43,40 @@ function updateZ(rng::AbstractRNG, Z::AbstractVector, alg::UniGrid, X::AbstractV
     Zn = deepcopy(Z)
     return updateZ!(rng, Zn, alg, X)
 end
+
+export UniformGrid
+struct UniformGrid{N, T} <: AbstractVector{T} 
+    proditer::Iterators.ProductIterator{NTuple{N, LinRange{T, Int64}}}
```

Comment:
Let's go with this for now, but we should be able to accept any iterable inputs.

Reply:
Yeah, this might not currently work with scalar inputs `x`. Will do some work on this today.

Reply:
I was more thinking about the fact that you specialized to the case where the inputs are `LinRange` only.

Reply:
Mostly because that is what the `InducingPoints` method currently returns. Would there be a use case for `UniformGrid` beyond being an output of `inducingpoints(UniGrid(...),...)`?

Reply:
Ah I see what you mean now, that makes sense. I was somehow already thinking about it in a broader scope, but in this application it makes sense.

Reply:
Yeah, happy to extend to broader use cases, but for now this specialization makes coding this fairly easy.